server/handlers/componetgenratorhelper.go

```diff
@@ -0,0 +1,647 @@
+		if err != nil {
+			return meshkitutils.ErrFileWalkDir(err, path)
+		}
+		if !info.IsDir() {
+			wg.Add(1)
+			go func(path string) {
+				defer wg.Done()
+				processFile(path, h, &mu, response, provider)
+			}(path)
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	wg.Wait()
+	return nil
+}
+
+func processFile(path string, h *Handler, mu *sync.Mutex, response *models.RegistryAPIResponse, provider models.Provider) {
+	if meshkitutils.IsZip(path) || meshkitutils.IsTarGz(path) {
+		newTempDir, err := os.MkdirTemp("", "nested-extracted-")
+		if err != nil {
+			incrementCounter(mu, &response.EntityCount.TotalErrCount)
+			h.log.Error(meshkitutils.ErrCreateDir(err, "Error creating nested temp directory"))
+			addUnsuccessfulEntry(path, response, meshkitutils.ErrCreateDir(err, "Error creating nested temp directory"), "")
+			return
+		}
+		defer os.RemoveAll(newTempDir)
+		if err := processUploadedFile(path, newTempDir, h, response, provider, false); err != nil {
+			incrementCounter(mu, &response.EntityCount.TotalErrCount)
+			h.log.Error(err)
+			addUnsuccessfulEntry(path, response, err, "")
+			return
+		}
+		return
+	}
+
+	content, err := os.ReadFile(path)
+	if err != nil {
+		incrementCounter(mu, &response.EntityCount.TotalErrCount)
+		h.log.Error(meshkitutils.ErrReadFile(err, path))
+		addUnsuccessfulEntry(path, response, meshkitutils.ErrReadFile(err, path), "")
+		return
+	}
+	processFileToRegistry(response, content, mu, path, h)
+}
+
+func processFileToRegistry(response *models.RegistryAPIResponse, content []byte, mu *sync.Mutex, path string, h *Handler) {
+	var jsonData interface{}
+	if err := yaml.Unmarshal(content, &jsonData); err == nil {
+		jsonData = utils.ConvertToJSONCompatible(jsonData)
+		convertedContent, err := json.Marshal(jsonData)
+		if err == nil {
+			content = convertedContent
+		}
+	}
+	entityType, err := meshkitutils.FindEntityType(content)
+	if err != nil {
+		incrementCounter(mu, &response.EntityCount.TotalErrCount)
+		addUnsuccessfulEntry(path, response, err, "")
+	}
+	if entityType != "" {
+		path, err := RegisterEntityForImport(content, entityType, h, response, mu)
+		if err != nil {
+			incrementCountersOnErr(mu, entityType, response)
+			h.log.Error(err)
+			addUnsuccessfulEntry(path, response, err, string(entityType))
+		} else {
+			if path != "" {
+				incrementCountersOnSuccess(mu, entityType, &response.EntityCount.CompCount, &response.EntityCount.RelCount, &response.EntityCount.ModelCount)
+				addSuccessfulEntry(content, entityType, response)
+			}
+
+		}
+	}
+}
+func addSuccessfulEntry(content []byte, entityType entity.EntityType, response *models.RegistryAPIResponse) {
+	// Helper function to check if a display name is already present
+	isDisplayNamePresent := func(displayName string) bool {
+		for _, name := range response.ModelName {
+			if name == displayName {
+				return true
+			}
+		}
+		return false
+	}
+
+	switch entityType {
```

Comment:
@Jougan-0 Recently, an `encoding` package was added in Meshkit, to have consistent error handling (returning Meshkit error) and the same logic for marshalling/unmarshalling and also supporting yaml and json data.

The existing Umarshal in Meshkit is deprecated, wherever we use utils.UnMarshal and Marshal we need to replace with encoding package.